swabbie-core/src/test/kotlin/com/netflix/spinnaker/swabbie/notifications/NotificationSenderTest.kt

```diff
@@ -0,0 +1,167 @@
+import com.netflix.spinnaker.swabbie.events.OwnerNotifiedEvent
+import com.netflix.spinnaker.swabbie.model.MarkedResource
+import com.netflix.spinnaker.swabbie.model.Summary
+import com.netflix.spinnaker.swabbie.model.WorkConfiguration
+import com.netflix.spinnaker.swabbie.repository.ResourceTrackingRepository
+import com.netflix.spinnaker.swabbie.test.InMemoryNotificationQueue
+import com.netflix.spinnaker.swabbie.test.TestResource
+import com.netflix.spinnaker.swabbie.test.WorkConfigurationTestHelper
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.reset
+import com.nhaarman.mockito_kotlin.whenever
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.doReturn
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.argumentCaptor
+import com.nhaarman.mockito_kotlin.times
+
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.context.ApplicationEventPublisher
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+import strikt.assertions.isNotEmpty
+import strikt.assertions.isNotNull
+import strikt.assertions.isNull
+import java.time.Clock
+
+object NotificationSenderTest {
+  private val notifier = mock<Notifier>()
+  private val clock = Clock.systemDefaultZone()
+  private var notificationQueue = InMemoryNotificationQueue()
+  private val resourceRepository = mock<ResourceTrackingRepository>()
+  private val applicationEventPublisher = mock<ApplicationEventPublisher>()
+
+  private val workConfiguration1 = WorkConfigurationTestHelper
+    .generateWorkConfiguration(namespace = "ns1", resourceType = "type1")
+  private val workConfiguration2 = WorkConfigurationTestHelper
+    .generateWorkConfiguration(namespace = "ns2", resourceType = "type2")
+
+  private val notificationService = NotificationSender(
+    lockingService = LockingService.NOOP(),
+    notifier = notifier,
+    clock = clock,
+    applicationEventPublisher = applicationEventPublisher,
+    resourceTrackingRepository = resourceRepository,
+    notificationQueue = notificationQueue,
+    registry = NoopRegistry(),
+    workConfigurations = listOf(workConfiguration1, workConfiguration2)
+  )
+
+  private val now = clock.millis()
+
+  @BeforeEach
+  fun setup() {
+    notificationService
+      .onDiscoveryUpCallback(
+        RemoteStatusChangedEvent(StatusChangeEvent(InstanceInfo.InstanceStatus.DOWN, InstanceInfo.InstanceStatus.UP)))
+  }
+
+  @AfterEach
+  fun cleanup() {
+    reset(
+      notifier,
+      resourceRepository,
+      applicationEventPublisher
+    )
+  }
+
+  @Test
+  fun `should not notify if there is nothing in the notification queue`() {
+    whenever(
+      resourceRepository.getMarkedResources()
+    ) doReturn listOf(
+      createMarkedResource(workConfiguration = workConfiguration2, id = "3", owner = "test@netflix.com")
+    )
+
```

Comment:
Don't you actually need to call the notifier here?

Reply:
Planning on revisiting this test